arch/arm/soc/nordic_nrf5/nrf52/power.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <soc_power.h>
+#include <nrf_power.h>
+
+/* System_OFF is deepest Power state available, On exiting from this
+ * state CPU including all peripherals reset
+ */
+static void _system_off(void)
+{
+	nrf_power_system_off();
+}
+
+void configure_ram_retention(void)
+{
+	/* Place Holder for RAM retainition */
```

Comment:
retention

Reply:
Does the kernel currently have support for retaining state over a full power cycle like the one we see with System OFF? If so it would be great if we could use it as an example

Reply:
No It does not support retaining the state on full cycle off. It loses context and it is starting from fresh boot. 
My intention was to start a patch, Later on we can enhance the functionality to save context in some memory location and restore the context on fresh boot.

Reply:
Currently on issuing System_Off state it just turns off. And on pressing GPIO it starts from fresh boot.